src/OrleansRuntime/Streams/QueueBalancer/DeploymentBasedQueueBalancer.cs

```diff
@@ -98,19 +98,33 @@ public class DeploymentBasedQueueBalancer : QueueBalancerBase, ISiloStatusListen
             await NotifyListeners();
         }
 
+        private async Task HandleSiloStatusChange()
+        {
+            while (!this.siloStatusOracle.CurrentStatus.IsTerminating())
+            {
+                try
+                {
+                    this.siloStatusSharedState = await this.siloStatusSharedState.NextAsync;
+                    HandleSiloStatusChange(this.siloStatusSharedState.State);
```

Comment:
I think ideal behavior is only process new state when there's a change. But this code seems to process state regardless whether it is new or not. Since `NextAsync=> this.NextCompletion.Task;`  while NextCompletion.Result  only get updated when there's new state. So most of the time the code keep processing the old NextCompletion.Result. Am I right?